tests/resources/generator/cell_geometry_model/model.c

```diff
@@ -0,0 +1,44 @@
+/* The content of this file was generated using the C profile of libCellML 0.2.0. */
+
+#include "model.h"
+
+#include <math.h>
+#include <stdlib.h>
+
+const char VERSION[] = "0.3.0";
+const char LIBCELLML_VERSION[] = "0.2.0";
+
+const size_t VARIABLE_COUNT = 4;
+
+const VariableInfo VARIABLE_INFO[] = {
+    {"L", "centimeter", "cell_geometry", CONSTANT},
+    {"rad", "centimeter", "cell_geometry", CONSTANT},
+    {"vcell", "microlitre", "cell_geometry", COMPUTED_CONSTANT},
+    {"vss", "microlitre", "cell_geometry", COMPUTED_CONSTANT}
+};
+
+double * createVariablesArray()
+{
+    return malloc(VARIABLE_COUNT*sizeof(double));
+}
+
+void deleteArray(double *array)
+{
+    free(array);
+}
+
+void initialiseVariables(double *variables)
+{
+    variables[0] = 0.01;
+    variables[1] = 0.0011;
+}
+
+void computeComputedConstants(double *variables)
+{
+    variables[2] = 1000.0*3.14*variables[1]*variables[1]*variables[0];
+    variables[3] = 0.02*variables[2];
+}
+
+void computeVariables(double *variables)
+{
+}
```

Comment:
This is an algebraic model (as opposed to a differential model), so no `computeRates` function is needed here. As for the the `computeVariables` function, it's empty because there are no variables to compute, but in theory there could be some in an algebraic model.

Reply:
So I guess my question was about clarity of the functions that are visible.  Are the functions based on the type of model that has been identified?  So if a model has been identified as algebraic then it will have methods such and such. And if it has been identified as differential it will have methods this and that.  This is an implicit definition correct?

Reply:
Correct. The functions are based on the type of model we are dealing with (i.e. algebraic or differential) AND on whether there are external variables.